engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -156,6 +156,7 @@
   private transient int nodeIndex = 0; // used for cycle validation
   private transient Stack<OperatorMeta> stack = new Stack<OperatorMeta>(); // used for cycle validation
   private transient Map<String, ArrayListMultimap<OutputPort<?>, InputPort<?>>> streamLinks = new HashMap<>();
+  private transient Map<String, Boolean> extraResources = new LinkedHashMap<>();
```

Comment:
Lets say one has created the jar on the fly and want to delete it to be deleted when the jar is consumes then one can chose to set true for this.
If one is using existing jars present in local system and include them on the fly, in that case user might not want to delete the existing jar.

This gives freedom to user to delete or not the original resource.